apps/api/src/user/service/user.service.ts

```diff
@@ -60,6 +114,66 @@ export class UserService {
     })
   }
 
+  async validateOtp(userId: User['id'], otp: string): Promise<User> {
+    const user = await this.getUserById(userId)
+    if (!user) {
+      throw new BadRequestException(`User ${userId} does not exist`)
+    }
+
+    const userEmailChange = await this.prisma.userEmailChange.findUnique({
+      where: {
+        otp: otp,
+        userId: userId,
+        createdOn: {
+          gt: new Date(new Date().getTime() - this.OTP_EXPIRY)
+        }
+      }
+    })
+
+    if (!userEmailChange) {
+      this.log.log(`OTP expired or invalid`)
+      throw new UnauthorizedException('Invalid or expired OTP')
+    }
+
+    await this.prisma.userEmailChange.delete({
+      where: {
+        userId: userId,
+        otp: otp
+      }
+    })
+
+    this.log.log(
+      `Changing email to ${userEmailChange.newEmail} for user ${userId}`
+    )
+    return await this.prisma.user.update({
+      where: {
+        id: userId
+      },
+      data: {
+        email: userEmailChange.newEmail
+      }
+    })
+  }
+
+  async resendOtp(userId: User['id']) {
```

Comment:
These would be changed as per the controller review

Reply:
Chnaged